openff/toolkit/topology/molecule.py

```diff
@@ -1223,16 +1223,17 @@ def _initialize_from_dict(self, molecule_dict):
         for iter_name, hierarchy_scheme_dict in molecule_dict[
             "hierarchy_schemes"
         ].items():
-            new_hier_scheme = self.add_hierarchy_scheme(
+            new_hier_scheme = HierarchyScheme(
```

Comment:
(not blocking) Good job on this whole section - It's possible that the molecule that we're deserializing here had its HierarchyElements defined, and then later had some `atom.metadata` info changed to that the original HierarchyElement definition was no longer current. In that case it would be important that we DON'T re-perceive the hierarchy during deserialization, and instead initialize the molecule to have the "outdated" hierarchy information. Nice work!
```suggestion
            # It's important that we do NOT call `add_hierarchy_scheme` here, since we 
            # need to deserialize these HierarchyElements exactly as they were serialized,
            # even if that conflicts with the current values in atom metadata. 
            new_hier_scheme = HierarchyScheme(
```

Reply:
Yeah, that was exactly the bug that I needed help from Matt with - it was leading to a duplication of the hierarchy schemes in tests!